content/doc/developer/extensibility/access-modifier-annotation.adoc

```diff
@@ -0,0 +1,58 @@
+---
+title: Access Modifier Annotation (@Restricted)
+layout: developer
+---
+
+All public classes, as well as their public methods and fields are available to plugins with a dependency on the component defining a class (Jenkins core or a plugin).
+Sometimes, these need to be public for technical reasons, but should not be available for other plugins to use.
+For example, Stapler web methods (`#doConfigSubmit`, `doCheckName`, `doFillJobItems`, and similar) need to be public to be invoked by Stapler, but they're often not stable API -- parameters may be added or removed depending on the needs of the UI.
+
+The access modifier annotation jenkinsdoc:component:access-modifier-annotation:org.kohsuke.accmod.Restricted[`@Restricted`] helps solving this problem, by preventing compliant plugins from calling methods annotated to be inaccessible by default.
```

Comment:
```suggestion
The access modifier annotation jenkinsdoc:component:access-modifier-annotation:org.kohsuke.accmod.Restricted[`@Restricted`] helps to solve this problem, by preventing compliant plugins from calling methods annotated to be inaccessible by default.
```

Reply:
```suggestion
The access modifier annotation jenkinsdoc:component:access-modifier-annotation:org.kohsuke.accmod.Restricted[`@Restricted`] helps solve this problem, by preventing compliant plugins from calling methods annotated to be inaccessible by default.
```
I think this could also be an alternative but either suggestion would work to adjust the grammar.

Reply:
Thanks to you both! The suggestion from @kmartens27 sounds better in my head, so I chose that one.